questions.md

```diff
@@ -93,6 +94,8 @@ Please keep in mind that this document is in a continuous state of flux, with ne
 - Why is it that, in swift, you can't have an array of, say, `Equatable`, ie, what's the reason for the compiler error "Protocol 'XXX' can only be used as a generic constraint because it has Self or associated type requirements"?
 - Are you familiar with functional-programming methods such as `map`, `filter`, and `reduce`? Can you explain their similarities and differences?
 - `Swift Standard Library`: What are `Generator`s, `Sequence`s and `Collection`s?
+- What problem do optionals solve?
+- Do you use optionals to handle state? Errors? Why?
```

Comment:
I would phrase it slightly differently, maybe:

`Do you use optionals to handle state? If so, how do you model Errors in an Optional context?